### PR TITLE
Use has_free_unbacked_symbols instead of bool(free_unbacked_symbols)

### PR DESCRIPTION
**Pull request by laithsakka**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #140027

with 20K features saves 20 seconds. 
257.021589517593-> 237.8304626941681
buck2 run @fbcode//mode/opt fbcode//torchrec/distributed/tests:pt2_compile_benchmark -- --num-features=2000

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Comment by laithsakka** [comment]
```
WIN: benchmark ('update_hint_regression', 'compile_time_instruction_count') failed, actual result 1696567937 is -3.05% lower than expected 1750000000 ±2.00% please update the expected results. 

please update all results that changed significantly, and not only the failed ones
WIN: benchmark ('sum_floordiv_regression', 'compile_time_instruction_count') failed, actual result 1096619420 is -5.46% lower than expected 1160000000 ±1.50% please update the expected results. 
```

**Review by ezyang — APPROVED** [review]

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -772,6 +773,14 @@ def has_free_symbols(val: IterateExprs) -> bool:
     return not all(e.is_number for e in _iterate_exprs(val))
 
 
+def has_free_unbacked_symbols(x: IterateExprs) -> bool:
+    """Faster version of bool(free_unbacked_symbols(val))"""
+    for s in free_symbols(x):
+        if symbol_is_type(s, (SymT.UNBACKED_INT, SymT.UNBACKED_FLOAT)):
+            return True
```
Comment (isuruf) [inline torch/fx/experimental/symbolic_shapes.py:780]: Following might be faster. Not sure.
```suggestion
    from sympy.core.traversal import iterargs
    for s in _iterate_exprs(x):
       for arg in iterargs(s):
          if isinstance(arg, sympy.Symbol) and symbol_is_type(arg, (SymT.UNBACKED_INT, SymT.UNBACKED_FLOAT)):
            return True
```

Comment (laithsakka) [reply torch/fx/experimental/symbolic_shapes.py:780]: i will try it out

**Pull request closed without merging**